File: src/typ/page.go
// page type
// @author xiangqian
// @date 10:52 2023/02/04
package typ

// Page 分页数据
type Page[T any] struct {
	Current int64   `json:"current"` // 当前页
	Size    uint8   `json:"size"`    // 页数量
	Total   int64   `json:"total"`   // 总数
	Pages   int64   `json:"pages"`   // 总页数
	Indexes []int64 `json:"indexes"` // 页数索引
	Data    []T     `json:"data"`    // 数据
}

File: src/main/java/org/xiangqian/note/model/Page.java
package org.xiangqian.note.model;

import lombok.Data;

import java.util.Collections;
import java.util.List;

/**
 * 分页信息
 *
 * @author xiangqian
 * @date 00:53 2022/06/12
 */
@Data
public class Page<T> {

    /**
     * 当前页
     */
    private Integer current;

    /**
     * 页数量
     */
    private Integer size;

    /**
     * 总数
     */
    private Integer total;

    /**
     * 数据
     */
    private List<T> data;

    public Page() {
        this(1, 10);
    }

    public Page(Integer current, Integer size) {
        this.current = current;
        this.size = size;
        this.total = 0;
        this.data = Collections.emptyList();
    }

}
